### ISSUE TITLE
Customized Locale throws exception while formatting 

### ISSUE THREAD

**Issue by SarathSantoshDamaraju** [iamkun/dayjs#746]
**Describe the bug**
I am trying to convert a time stamp to a required format 'LT'. 
```js
dayjs(1575872723701);

/* Returns this object
{
$D: 19
$H: 12
$L: undefined
$M: 10
$W: 2
$d: Tue Nov 19 2019 12:26:03 GMT+0530 (India Standard Time) {}
$m: 26
$ms: 136
$s: 3
$y: 2019
}
*/
```
and all the APIs like `add`, `subtract`, `date` works except `format`. 
```js
dayjs(1575872723701).format() 
```
 returns **Uncaught TypeError: Cannot read property '10' of undefined**

#### My usage

- I have created a service file with all the plugin initializations of dayJs and importing it into the application
```js
// services/dayjs.js
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import relativeTime from 'dayjs/plugin/relativeTime';
import localizedFormat from 'dayjs/plugin/localizedFormat'

dayjs.extend(advancedFormat);
dayjs.extend(relativeTime);
dayjs.extend(localizedFormat)

export default dayjs;

// somefile.js
import dayjs from './services/dayjs.js';

...
dayjs(1575872723701).format()  // throws error
```

- I dont see any issue with other plugins and APIs except `format`.

- I also set the locale of the `/services/dayjs.js` based on the locale sent. if not set to the below default config

```js
import dayjs from './services/dayjs.js';

if(...){
const locale = {
      relativeTime: {
        name: 'en',
        future: '%s',
        past: '%s',
        s: function () {
          return 'now';
        },
        m: 'a minute ago',
        mm: '%d minutes ago',
        h: 'an hour',
        hh: '%d hours', // e.g. 2 hours, %d been replaced with 2
        d: 'a day',
        dd: '%d days',
        M: 'a month',
        MM: '%d months',
        y: 'a year',
        yy: '%d years'
      },
      ordinal: (n) => {
        const s = ['th', 'st', 'nd', 'rd']
        const v = n % 100
        return `[${n}${(s[(v - 20) % 10] || s[v] || s[0])}]`
      }
    }

    dayjs.locale(locale)
}
```

**Would this be causing any error ?**

**Expected behavior**
Should ideally format to the provided option

**Information**
 - Day.js Version: 1.8.17
 - OS: MAC OS 10.15.1
 - Browser: Chrome Version 78.0.3904.108 and all major browsers
 - Time zone: GMT + 5.30

**Comment by SarathSantoshDamaraju** [comment]
Fiddle for the same https://jsfiddle.net/5o4pwbtc/

**Comment by SarathSantoshDamaraju** [comment]
Found the solution. Setting locale for 2nd time made it work. //2

Sample code
```js
var dayjs = require("dayjs")
var relativeTime = require('dayjs/plugin/relativeTime')

dayjs.extend(relativeTime)

const locale = {
  name: 'en',
  weekdays: 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_'),
  weekdaysShort: 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_'),
  weekdaysMin: 'Su_Mo_Tu_We_Th_Fr_Sa'.split('_'),
  months: 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_'),
  monthsShort: 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_'),
  weekStart: 1,
  relativeTime: {
    future: '%s',
    past: '%s',
    s: 'now',
    m: 'a minute',
    mm: '%d minutes',
    h: 'an hour',
    hh: '%d hours',
    d: 'a day',
    dd: '%d days',
    M: 'a month',
    MM: '%d months',
    y: 'a year',
    yy: '%d years'
  }
}

dayjs.locale(locale, null, true) //1
dayjs.locale(locale) //2
dayjs().format()
```

Clarifications needed:

1. Setting custom locale expects atleast the above given keys for it to work properly. Cannot update it like moment with partial locale configs.

2. What is the need to use `.locale()` twice ?

CC: @iamkun

**Comment by iamkun** [comment]
![image](https://user-images.githubusercontent.com/17680888/70522368-f4c99380-1b7b-11ea-8126-8a7685309e1e.png)
https://runkit.com/embed/xlq6nc5uoylk

No error here. Is this problem still exist?

**Comment by SarathSantoshDamaraju** [comment]
I took a workd around. But

I faced this while implementing the custom locale. 

If I want to extend the `en` locale just with `relativeTime: {}` then the error happens i assume custome locale expects to declare `weekdays`,  `weekdaysShort` and other related keys as mentioned in the below comment.
https://github.com/iamkun/dayjs/issues/746#issuecomment-563927090

This API dosent align with moment implementation.
ref: https://runkit.com/embed/6abjer62rzj0
